app/service/data_svc.py

```diff
@@ -201,3 +198,22 @@ def __init__(self, dao):
 
     async def update(self, table, key, value, data):
         await self.dao.update(table, key, value, data)
+
+
+    """ SUPPORT """
+    async def _create_attack(self, technique, tactic):
+        entry = await self.dao.get('core_attack', dict(attack_id=technique['attack_id']))
```

Comment:
I like what this function does, I just don't like it's implementation - I think we can simplify it a bit. 

for the first part (lines 205-209) I think we can just insert (blindly) and retrieve the entry object. this should piggyback on the UNIQUE clause of the core_attack table "UNIQUE(attack_id)" - which we should ensure is updated to: UNIQUE(attack_id) ON CONFLICT IGNORE. 

then, in the next section (lines 211-217) we should attempt to remove the if/else 'null' check.

Reply:
Cleaned this up.